<launch>
    <!-- Set robot dimensions and load urdf and sdf -->

    <arg name="base_radius" default="0.1"/>
    <arg name="ee_radius" default="0.1"/>
    <arg name="upper_length" default="0.3"/>
    <arg name="lower_length" default="0.4"/>

    <param
        name="robot_description"
    command="xacro '$(find cga_robotics_ros)/model/serial/ros.urdf.xacro'"
    />

    <!-- Start hardware interface -->

    <rosparam
        file="$(find cga_robotics_ros)/config/serial_gazebo_control_gains.yaml"
        command="load"
    />
    <node
        name="serial_hardware_interface"
        pkg="cga_robotics_ros"
        type="serial_hardware_interface"
        output="screen"
    />

    <!-- Start ros_control -->

    <rosparam
        file="$(find cga_robotics_ros)/config/serial_controllers.yaml"
        command="load"
    />
    <node
        name="controller_spawner"
        pkg="controller_manager"
        type="spawner"
        respawn="false"
        args="theta_controller gripper_controller joint_state_controller"
    />

</launch>
